### PR TITLE
Define custom theme

### PR DESCRIPTION
**Pull request by mirland**
### Description:
The current theme was divided into multiple files to handle the different app properties:
- [AppDims](https://github.com/xmartlabs/gong/blob/feature/add-custom-theme/app/src/main/java/com/xmartlabs/gong/ui/theme/AppDims.kt) contains the app common dims, for example, common text size or common margins that will be used in the app.
Then, an example mode, the app defines 2 dimensions sets (2 AppDims instances), [one for regular decices](https://github.com/xmartlabs/gong/blob/feature/add-custom-theme/app/src/main/java/com/xmartlabs/gong/ui/theme/AppDims.kt#L45) and the [other for small devices](https://github.com/xmartlabs/gong/blob/feature/add-custom-theme/app/src/main/java/com/xmartlabs/gong/ui/theme/AppDims.kt#L61) (a small device is a device that the screen width is lower than 300dp)
- [AppColors](https://github.com/xmartlabs/gong/blob/feature/add-custom-theme/app/src/main/java/com/xmartlabs/gong/ui/theme/AppColors.kt) defines the app colors.
[AppColors](https://github.com/xmartlabs/gong/blob/feature/add-custom-theme/app/src/main/java/com/xmartlabs/gong/ui/theme/AppColors.kt#L15) is a class that wraps material colors but it also gives the ability to define custom colors.
For example, in this case, we defined `linkColor` and `subtitleTextColor` colors that will be used in the app.
`AppCustomColors` is a private class the defines the color instances that will be used by `AppColors`, it's a private class because you shouldn't use it as a color, you cannot use `AppColors.RED` because it depends on the app status. If you're in dark mode or in light mode, the app colors should be consistent with that. However, sometimes you need to use some colors, for example, to define something in a library or something like that, if that's the case you have to define a  `Color` extension in this way:
```kotlin
@Stable
val Color.Companion.WhiteLilac
  get() = AppCustomColors.WHITE_LILAC
```
The app defines 2 `AppColors` instances, one for light mode and the other for dark mode.
- [AppTypography](https://github.com/xmartlabs/gong/blob/feature/add-custom-theme/app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt) follows the same idea, this class wraps material  Typography, but it provides the ability to define custom Typography properties. In that case, we defined `primaryButton` and `linkButon` so if you have a link button you can use it!

With all these classes we define the [`AppTheme`](https://github.com/xmartlabs/gong/blob/feature/add-custom-theme/app/src/main/java/com/xmartlabs/gong/ui/theme/AppTheme.kt), a theme that wraps material theme, but it has all the custom properties.
 In this case, we have `AppTheme.colors`, `AppTheme.dims`  and `AppTheme.typography`
 So if you are using `AppTheme` and you have something like that:
 ```
  Text("Team Android", style = AppTheme.typography.primaryButton)
```
It will use the colors depending your system current status (light mode or dark mode) and the text size based on your phone specs

### CODE REVIEW

**Review by f7deleon — COMMENTED** [review]
can you add a description for this change? i thought it was ok how was initially planned since the idea was to have multiple themes on the app but maybe you and sante discussed this more in depth the change

**Comment by mirland** [comment]
> can you add a description for this change? i thought it was ok how was initially planned since the idea was to have multiple themes on the app but maybe you and sante discussed this more in depth the change

Using this implementation, the app can manage multiple themes, there's no problem with that, but `AppTheme` contains the current app theme. 
I used the Material idea, in one place you have to define what theme are you using in that moment

**Review by f7deleon — APPROVED** [review]

File: app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    get() = materialTypography.button
+
+  val caption: TextStyle
+    get() = materialTypography.caption
+
+  val overline: TextStyle
+    get() = materialTypography.overline
+}
+
+fun detDefaultAppTypography() = getAppTypography(defaultAppDims(), defaultAppColors())
```
Comment (sante) [inline app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt:66]: Why det?

File: app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+fun getAppTypography(dims: AppDims, colors: AppColors): AppTypography {
+  val regularTextSize = TextStyle(
+      fontFamily = AppFontFamilies.Roboto,
+      fontWeight = FontWeight.Normal,
+      fontSize = dims.textSizeRegular,
+      letterSpacing = (-1.5).sp,
+  )
```
Comment (sante) [inline app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt:74]: This should be `regularTextStyle`

File: app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt (hunk 1 above)
Comment (mirland) [reply app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt:66]: typo of get

File: app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+fun defaultAppTypography() = appTypography(defaultAppDims(), defaultAppColors())
+
+fun appTypography(dims: AppDims, colors: AppColors): AppTypography {
+  val baseTextStyle = TextStyle(
+      fontFamily = AppFontFamilies.Roboto,
+      fontWeight = FontWeight.Normal,
+      fontSize = dims.textSizeRegular,
+      letterSpacing = (-1.5).sp,
+  )
+
```
Comment (sante) [inline app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt:75]:
```suggestion

fun defaultAppTypography() = appTypography()

fun appTypography(dims: AppDims = defaultAppDims(), colors: AppColors = defaultAppColors()): AppTypography {
  val baseTextStyle = TextStyle(
      fontFamily = AppFontFamilies.Roboto,
      fontWeight = FontWeight.Normal,
      fontSize = dims.textSizeRegular,
      letterSpacing = (-1.5).sp,
  )

```

Maybe it's easier to create more typographies this way, maybe we want to change just the colors or just the dims

**Review by sante — APPROVED** [review]

**Pull request merged**